### DISCUSSION TITLE
Bumpy experience with converting default PostgreSQL DateTime to Date

### DISCUSSION THREAD

**Discussion by alexandrulesi** [prisma/prisma#15686] [Q&A] [answered]
We're storing create and update DateTimes for most of the tables in our db. We started off using the default `DateTime` type suggested by Prisma, without giving it a second thought. This translates to TIMESTAMP(3) for PostgreSQL. The stored DateTimes look like this in our db:

![image](https://user-images.githubusercontent.com/37101969/194274914-84bf54d7-d9be-4e5d-ba8e-25a0b1589cfa.png)

They all get converted to UTC timezone and do not specify the timezone themselves anymore. We are using an auto-generated Postgraphile Graphql api on top of it, which retrieves the DateTimes as these strings as is. We're not able to query it in a different format, say milliseconds. 

If we try to create Dates in our FE from them these dates will automatically be assumed to be in the local timezone, and not UTC. To specify them as UTC we either need to manually change the string to append a `Z`, making them valid ISO strings, or to shift the time zone manually. Obviously we'd like to avoid that. 

We ended up changing all the types in our db to DateTime @db.Timestamptz(3).

Maybe this is fine but we just got a little surprised to having such issues using the default DateTime. I'd expect to be able to do basic things, like simply creating dates from string, without having to first make sure I picked the right type of DateTime and not just the default.  I figured I'd open a discussion here in case there is something we got wrong or this is of interest to the Prisma team.

**Comment by nurul3101** [comment, marked as answer]
Hi @alexandrulesi 👋 
Thank you for letting us know about the feedback on using the default DateTime type, I will share this with our team and also discuss the possibility of highlighting this case better in our docs.

**Reply by nurul3101** [reply]
@alexandrulesi I have opened a Docs Issue for this: [#3861](https://github.com/prisma/docs/issues/3861)
Thanks for providing the feedback.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
